docs/dev_guide/contents/extending_fido.rst

```diff
@@ -0,0 +1,501 @@
+
+``baseurl`` becomes ``r'https://some-domain.com/%Y/%m/%d/satname_(\d){2}_(\d){1}_(\d){12}_(\d){2}\.fits'``.
+
+Note all variables in the filename are converted to regex that will match any possible value for it.
+A character enclosed within ``()`` followed by a number enclosed within ``{}`` is used to match the specified number of occurences of that special sequence.
+For example, ``%y%m%d%H%M%S`` is a twelve digit variable (with 2 digits for each item) and thus represented by ``r'(\d){12}'``.
+Note that ``\`` is used to escape the special character ``.``.
+
+``pattern`` becomes ``'{}/{year:4d}/{month:2d}{day:2d}/satname_{SatelliteNumber:2d}_{Level:1d}_{:6d}{hour:2d}{minute:2d}{second:2d}_{:2d}.fits'``.
+Note the sole purpose of ``pattern`` is to extract the information from matched URL, using :func:`~sunpy.extern.parse.parse`.
+So the desired key names for returned dictionary should be written in the ``pattern`` within ``{}``, and they should match with the ``attr.__name__``.
+
+``register_values()`` can be written as:
+
+.. code-block:: python
+
+    @classmethod
+    def register_values(cls):
+
+        from sunpy.net import attrs
+        adict = {
+        attrs.Instrument: [("SatName", "The description of Instrument")],
+        attrs.Physobs: [('some_physobs', 'Phsyobs description')],
+        attrs.Source: [('some_source', 'Source description')],
+        attrs.Provider: [('some_provider', 'Provider description')],
+        attrs.Level: [("1", "Level 1 data"), ("2", "Level 2 data")],
+        attrs.SatelliteNumber: [("16", "Describe it"), ("17", "Describe it")]
+        }
+
+        return adict
+
+
+.. _new_full_client:
+
+Writing a "full" client
+=======================
+
+In this section we will describe how to build a "full" Fido client.
+You should write a new "full" client if the data you are accessing can not be accessed via a URL template, for instance if you hit a web API with a query to return results for a search.
+
+A new Fido client contains three major components:
+
+* A subclass of `~sunpy.net.base_client.BaseClient` which implements ``search``, ``fetch``, and ``_is_datasource_for``.
+* Zero or more new `~sunpy.net.attr.Attr` classes to specify search parameters unique to your data source.
+* An instance of `~sunpy.net.attr.AttrWalker` which can be used to walk the tree of `~sunpy.net.attr.Attr` instances and convert them into a form useful to your client's search method.
+
+Search Attrs
+------------
+
+As described in `~sunpy.net.attr` the attr system allows the construction of complex queries by the user.
+To make these complex queries easily processable by the clients the ``AttrWalker`` converts these into a set of queries which can be processed separately.
+It does this by converting the input query to a set of queries which are ORed, but are complete queries.
```

Comment:
maybe this is just me not knowing this but is `ORed` like OR-ed?

Reply:
yes? I am not really sure what the best formatting for this is.